api/api-facade/api-http/src/main/java/org/eclipse/dirigible/api/v3/http/HttpClientFacade.java

```diff
@@ -206,13 +207,20 @@ public static HttpClientResponse processHttpClientResponse(CloseableHttpResponse
 			if (entity != null && entity.getContent() != null) {
 				byte[] content = IOUtils.toByteArray(entity.getContent());
 
-				if ((ContentType.getOrDefault(entity).getMimeType().equals(ContentType.TEXT_PLAIN.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.TEXT_HTML.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.TEXT_XML.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.APPLICATION_JSON.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.APPLICATION_ATOM_XML.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.APPLICATION_XML.getMimeType())
-						|| ContentType.getOrDefault(entity).getMimeType().equals(ContentType.APPLICATION_XHTML_XML.getMimeType())) && (!binary)) {
+				 HashMap<String, Boolean> textMimeTypes = new HashMap<String, Boolean>();
+						textMimeTypes.put("application/CSV", true);
+						textMimeTypes.put("application/csv", true);
+						textMimeTypes.put("text/csv", true);
+						textMimeTypes.put(ContentType.TEXT_PLAIN.getMimeType(), true);
+						textMimeTypes.put(ContentType.TEXT_HTML.getMimeType(), true);
+						textMimeTypes.put(ContentType.TEXT_XML.getMimeType(), true);
+						textMimeTypes.put(ContentType.APPLICATION_JSON.getMimeType(), true);
+						textMimeTypes.put(ContentType.APPLICATION_ATOM_XML.getMimeType(), true);
+						textMimeTypes.put(ContentType.APPLICATION_XML.getMimeType(), true);
+						textMimeTypes.put(ContentType.APPLICATION_XHTML_XML.getMimeType(), true);
+				final boolean isTextType = textMimeTypes.get(ContentType.getOrDefault(entity).getMimeType()) != null;
+
+				if (isTextType && (!binary)) {
```

Comment:
Rename `isTextType` to `isSupportedTextType` and invert the condition to: `if (!binary && isSupportedTextType)`

Reply:
Fixed in last commit